AppCenter/AppCenter/MSACNSLogQueueManager.m

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSACNSLogQueueManager.h"
+#import <dispatch/dispatch.h>
+
+@implementation MSACNSLogQueueManager
```

Comment:
Why do we need a new singleton class for keeping one serialQueue which is only used in `MSACLogger`?
Can we just add a static variable to `MSACLogger`, eg `static dispatch_queue_t loggerDispatchQueue`. And instantiate in `+ (void)initialize`.

Reply:
Sure we can have it as a static variable on the class level, but I went with the singleton class just in case we later had to implement NLogging in different classes/modules to use the same serial queue instead of having to create another serial queue with the same purpose.

Reply:
There shouldn't be any places using NSLog other than MSACLogger. All logging should go through MSACLogger. The idea is to being able to replace the log handler which is NSLog by default. There is a public API for that `AppCenter.logHandler`. So if this dispatcher is only for NSLog, I believe it should be in MSACLogger. 
If for some special reason we would need to implement another logger or use NSLog in other places, it wouldn't be difficult to create a class for the dispatch queue.
For now, I don't think creating code just in case it may be used is a good idea.


Btw, I noticed we are using NSLog instead of MSACLogger in one place:
https://github.com/microsoft/appcenter-sdk-apple/blob/0361dbd7c98e29af468fd36db1441826ef55aa39/AppCenter/AppCenter/MSACAppCenter.m#L224

You can fix it in your PR.